pkg/kademlia/dialer.go

```diff
@@ -47,29 +48,37 @@ func (dialer *Dialer) Close() error {
 }
 
 // Lookup queries ask about find, and also sends information about self.
-func (dialer *Dialer) Lookup(ctx context.Context, self pb.Node, ask pb.Node, find pb.Node) (_ []*pb.Node, err error) {
+// If self is nil, pingback will be false.
+func (dialer *Dialer) Lookup(ctx context.Context, self *pb.Node, ask pb.Node, find storj.NodeID, limit int) (_ []*pb.Node, err error) {
 	defer mon.Task()(&ctx)(&err)
 	if !dialer.limit.Lock() {
 		return nil, context.Canceled
 	}
 	defer dialer.limit.Unlock()
 
+	req := pb.QueryRequest{
+		Limit:  int64(limit),
+		Target: &pb.Node{Id: find}, // TODO: should not be a Node protobuf!
```

Comment:
What should it be if not a node protobuf?

Reply:
it should just be a node id. a node also includes an address and so on. all we care about for the target is the id.